ensime_shared/ensime.py

```diff
@@ -49,41 +50,46 @@ class EnsimeClient(TypecheckHandler, DebuggerClient, ProtocolHandler):
     which stores the a handler per response type.
     """
 
-    def __init__(self, vim, launcher, config_path):  # noqa: C901 FIXME
-        def setup_vim():
-            """Set up vim and execute global commands."""
-            self.vim = vim
-            if not int(self.vim_eval("exists_enerrorstyle")):
-                self.vim_command("set_enerrorstyle")
-            self.vim_command("highlight_enerror")
-            self.vim_command("set_updatetime")
-            self.vim_command("set_ensime_completion")
-            self.vim.command(
-                "autocmd FileType package_info nnoremap <buffer> <Space> :call EnPackageDecl()<CR>")
-            self.vim.command("autocmd FileType package_info setlocal splitright")
-            super(EnsimeClient, self).__init__()
-
-        def setup_logger_and_paths():
-            """Set up paths and logger."""
-            osp = os.path
-            self.config_path = osp.abspath(config_path)
-            config_dirname = osp.dirname(self.config_path)
-            self.ensime_cache = osp.join(config_dirname, ".ensime_cache")
-            self.log_dir = self.ensime_cache
-            if not osp.isdir(self.ensime_cache):
+    def __init__(self, editor, vim, launcher):  # noqa: C901
```

Comment:
I can't *quite* jettison the `vim` attribute yet, the remaining usage is for `disable_plugin` which is too complicated to deal with in the course of these changes, I've dived into it a bit for #294 and will continue later.